lib/mayaUsd/nodes/proxyShapeBase.h

```diff
@@ -379,6 +379,10 @@ class MayaUsdProxyShapeBase
 
     // Keep track of the incoming layers
     std::set<std::string> _incomingLayers;
+
+public:
+    // Counter for the number of times compute is re-entered
+    static std::atomic<int> in_compute;
```

Comment:
Because this member is static, it will be used by all proxy shape nodes.  What if there are more than one proxy shape node in the scene?  Can it happen that both would enter compute() in parallel, and that we could drop a notification for one or more stages?

Reply:
Since in_compute is atomic, parallel calls to compute() should not be an issue. g_StageMap and g_StagesSubject are also global so I think the important is that onStageSet work be done only once.